libraries/AP_Proximity/AP_Proximity_LD06.h

```diff
@@ -0,0 +1,104 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*
+ * ArduPilot device driver for Inno-Maker LD06 LiDAR
+ *
+ * ALL INFORMATION REGARDING PROTOCOL WAS DERIVED FROM LD06 DATASHEET:
+ *
+ * http://wiki.inno-maker.com/display/HOMEPAGE/LD06?preview=/6949506/6949511/LDROBOT_LD06_Development%20manual_v1.0_en.pdf
+ *
+ * Author: Adithya Patil, Georgia Institute of Technology
+ * Based on the SLAMTEC RPLiDAR code written by Steven Josefs, IAV GmbH
+ *
+ */
+
+
+#pragma once
+
+#include "AP_Proximity.h"
+
+#ifndef AP_PROXIMITY_LD06_ENABLED
+#define AP_PROXIMITY_LD06_ENABLED HAL_PROXIMITY_ENABLED
+#endif
+
+#if HAL_PROXIMITY_ENABLED && AP_PROXIMITY_LD06_ENABLED
+#include "AP_Proximity_Backend_Serial.h"
+
+#define MESSAGE_LENGTH      47
+#define LD_START_CHAR       0x54
+#define PROXIMITY_LD06_TIMEOUT_MS 50
+
+// Indices in data array where each value starts being recorded
+// See comment below about data payload for more info about formatting
+#define START_BEGIN_CHARACTER       0
+#define START_DATA_LENGTH           1
+#define START_RADAR_SPEED           2
+#define START_BEGIN_ANGLE           4
+#define START_PAYLOAD               6
+#define START_END_ANGLE             42
+#define START_CHECK_SUM             46
+#define MEASUREMENT_PAYLOAD_LENGTH  3
+#define PAYLOAD_COUNT               12
```

Comment:
Nobody else in the system cares about these constants - they should be in the cpp